packages/jest-haste-map/src/index.js

```diff
@@ -230,6 +231,18 @@ class HasteMap {
     });
   }
 
+  matchGlobs(globs: Array<Glob>, rootDir: Path): Promise<Array<Path>> {
+    return this.build().then(hasteMap => {
+      const files = [];
+      for (const file in hasteMap.files) {
+        if (multimatch([path.relative(rootDir, file)], globs).length) {
```

Comment:
Just for the sake of comparison, can you, on www, try to call `matchGlobs` and `matchFiles` with the same pattern and compare the perf over like 10 runs or so? I have an unsettling feeling that this is going to be super costly (like 10x slower) and we might have to become creative. We run this code a lot on very big arrays and if its too slow we can't use globs.